os/board/rtl8730e/src/rtl8730e_ist415.c

```diff
@@ -0,0 +1,190 @@
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <debug.h>
+#include <assert.h>
+#include <errno.h>
+#include <tinyara/irq.h>
+#include <tinyara/gpio.h>
+#include <tinyara/input/touchscreen.h>
+#include <tinyara/input/ist415.h>
+#include "objects.h"
+#include "gpio_irq_api.h"
+#include "PinNames.h"
+#include "gpio_api.h"
+
+#define PIN_LOW		0
+#define PIN_HIGH	1
+#define TOUCH_DEV_PATH "/dev/touch0"
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+/* i2c config */
+#if CONFIG_RTL8730E_BOARD_REVISION >= 6
+#define IST415_I2C_PORT		0
+#else
+#define IST415_I2C_PORT		1
+#endif
+
+#define IST415_I2C_FREQ		100000
+#define IST415_I2C_ADDRLEN	7
+#define IST415_I2C_ADDR		(0xA0 >> 1) 
+
+/* pin config */
+#define IST415_GPIO_RESET_PIN	PA_5
+#if CONFIG_RTL8730E_BOARD_REVISION >= 6
+#define IST415_GPIO_I2C_PIN		PA_4
+#else
+#define IST415_GPIO_I2C_PIN		PA_2
+#endif
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+struct rtl8730e_ist415_s {
+	gpio_irq_t data_ready;
+};
+
+struct rtl8730e_ist415_s g_rtl8730e_ist415_priv0;
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static void rtl8730e_ist415_enable_irq(void);
+static void rtl8730e_ist415_disable_irq(void);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct ist415_dev_s g_ist415_dev0 = {
+	.i2c = NULL,
+	.i2c_config = {
+		.frequency = IST415_I2C_FREQ,
+		.address = IST415_I2C_ADDR,
+		.addrlen = IST415_I2C_ADDRLEN,
+	},
+	.int_pending = false,
+	.ops = &(struct ist415_ops_s){
+		.irq_enable = rtl8730e_ist415_enable_irq,
+		.irq_disable = rtl8730e_ist415_disable_irq,
+	},
+	.priv = &g_rtl8730e_ist415_priv0,
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static void rtl8730e_ist415_irq_handler(uint32_t id, gpio_irq_event event)
+{
+	/* we are using level based interrupt trigger here,
+	 * so, we have to disable this particular gpio interrupt
+	 * until we finish this particular interrupt related work
+	 * in the HPWORK thread
+	 */
+	struct ist415_dev_s *dev;
+	if (id == 0) {
+		dev =  &g_ist415_dev0;
+	}
+	struct rtl8730e_ist415_s *priv = dev->priv;
+	gpio_irq_disable((gpio_irq_t *)&priv->data_ready);
```

Comment:
I am not sure but as I know, it should not disable entire of interrupt.
```
if (touh_down_mode()) {
	diable_interrupt;
	do_something;
	enable_interrupt;
}
```
Because 'up' event happened only once, so it is required to prevent lose up event during process touch.

Reply:
done.